libraries/AP_Scripting/lua_bindings.cpp

```diff
@@ -45,10 +47,56 @@ static int lua_micros(lua_State *L) {
     return 1;
 }
 
+static int lua_get_i2c_device(lua_State *L) {
+
+    const int args = lua_gettop(L);
+    if (args < 2) {
+        return luaL_argerror(L, args, "require i2c bus and address");
+    }
+    if (args > 4) {
+        return luaL_argerror(L, args, "too many arguments");
+    }
+
+    const lua_Integer bus_in = luaL_checkinteger(L, 1);
+    luaL_argcheck(L, ((bus_in >= 0) && (bus_in <= 4)), 1, "bus out of range");
+    const uint8_t bus = static_cast<uint8_t>(bus_in);
+
+    const lua_Integer address_in = luaL_checkinteger(L, 2);
+    luaL_argcheck(L, ((address_in >= 0) && (address_in <= 128)), 2, "address out of range");
+    const uint8_t address = static_cast<uint8_t>(address_in);
+
+    // optional arguments, use the same defaults as the hal get_device function
+    uint32_t bus_clock = 400000;
+    bool use_smbus = false;
+
+    if (args > 2) {
+        bus_clock = coerce_to_uint32_t(L, 3);
+
+        if (args > 3) {
+            use_smbus = static_cast<bool>(lua_toboolean(L, 4));
+        }
+    }
+
+    if (AP::scripting()->num_i2c_devices >= SCRIPTING_MAX_NUM_I2C_DEVICE) {
+        return luaL_argerror(L, 1, "no i2c devices available");;
+    }
+
+    AP::scripting()->_i2c_dev[AP::scripting()->num_i2c_devices] = new AP_HAL::OwnPtr<AP_HAL::I2CDevice>;
+    *AP::scripting()->_i2c_dev[AP::scripting()->num_i2c_devices] = std::move(hal.i2c_mgr->get_device(bus, address, bus_clock, use_smbus));
```

Comment:
The device you get could be `nullptr` you need to actually check that it isn't a nullptr here. If it is we should probably be declaring an error.

Reply:
added a check for both the own pointer being null and the pointer in the own pointer being null

Reply:
You have to check that the OwnPtr isn't null here, not after you've dereferenced it

Reply:
fixed